### PR TITLE
#333/Reconnect the client socket after the user logout

### LINKED ISSUE #333: Reconnect the client socket after the user logout
When a user logout, the server [disconnects the client socket](https://github.com/42AGV/ft_transcendence/blob/b389665dffe3a9383956da02893551cc46fa494f/transcendence-app/src/auth/auth.controller.ts#L67). If they log in again without refreshing the webapp, they can't send chat messages.

Possible solutions:
1. Manually [connect](https://socket.io/docs/v4/client-api/#socketconnect) the socket after the user logs in.
2. Listen to the ['disconnect'](https://socket.io/docs/v4/client-api/#event-disconnect) event on the client side and reconnect automatically.

### PR DESCRIPTION
**Pull request by gbudau**
When a user logs in, we connect the socket, and when they log out, we [disconnect all the sockets with that `sessionId`](https://github.com/42AGV/ft_transcendence/blob/b389665dffe3a9383956da02893551cc46fa494f/transcendence-app/src/auth/auth.controller.ts#L67) .

fix #333

### CODE REVIEW

File: webapp/src/shared/context/auth-context.tsx [hunk 1]
```diff
@@ -46,6 +47,12 @@ export const AuthProvider = ({ children }: { children: ReactNode }) => {
     setAuthUser(data);
   }, [data, isDataLoading]);
 
+  useEffect(() => {
+    if (authUser) {
+      socket.connect();
+    }
```
Comment (gbudau) [inline webapp/src/shared/context/auth-context.tsx:53]: We set `autoConnect` to false and manually connect when the user logs in.

**Review by fizcamposancos — APPROVED** [review]
Nice!!

**Review by Damrod — APPROVED** [review]
Thanks for this feature!
LGTM!

**Comment by Damrod** [comment]
I have noticed that this is still happening with a user that just joins an existing chatroom. Please, take a look, or let me show you.

**Comment by gbudau** [comment]
Thanks!

**Pull request merged**